lib/elixir/pages/Guards.md

```diff
@@ -0,0 +1,226 @@
+
+An important thing to keep in mind is that errors in guards do not result in a runtime error, but in the erroring guard fail. For example, the `length/1` function only works with lists, and if we use it on anything else it fails:
+
+```elixir
+iex> length("hello")
+** (ArgumentError) argument error
+```
+
+However, when used in guards, it simply makes the corresponding clause fail:
+
+```elixir
+iex> case "hello" do
+...>   something when length(something) > 0 ->
+...>     :length_worked
+...>   _anything_else ->
+...>     :length_failed
+...> end
+:length_failed
+```
+
+In many cases, we can take advantage of this: in the code above, for example, we can use `length/1` to both check that the given thing is a list *and* check some properties of its length (instead of using `is_list(something) and length(something) > 0`).
+
+## Expressions in guard clauses
+
+Not all expressions are allowed in guard clauses, but only a handful of them. This is a deliberate choice: only a predefined set of side-effect-free functions are allowed. This way, Elixir (and Erlang) can make sure that nothing bad happens while executing guards and no mutations happen anywhere. This behaviour is also coherent with pattern match, which is a naturally a side-effect-free operation.
+
+The following is a comprehensive list of all expressions allowed in guards:
+
+  * comparison operators (`==`, `!=`, `===`, `!==`, `>`, `>=`, `<`, `<=`)
+  * strictly boolean operators (`and`, `or`, `not`) (the `&&`, `||`, and `!` sibling operators are not allowed as they're not *strictly* boolean - meaning they don't require both sides to be booleans)
+  * arithmetic binary operators (`+`, `-`, `*`, `/`)
+  * arithmetic unary operators (`+`, `-`)
+  * binary concatenation operator (`<>`)
+  * `in` operator (as long as the right-hand side is a list or a range)
+  * the following "type-check" functions (all documented in the `Kernel` module):
+    * `is_atom/1`
+    * `is_binary/1`
+    * `is_bitstring/1`
+    * `is_boolean/1`
+    * `is_float/1`
+    * `is_function/1`
+    * `is_function/2`
+    * `is_integer/1`
+    * `is_list/1`
+    * `is_map/1`
+    * `is_nil/1`
+    * `is_number/1`
+    * `is_pid/1`
+    * `is_port/1`
+    * `is_reference/1`
+    * `is_tuple/1`
+  * the following guard-friendly functions (all documented in the `Kernel` module):
+    * `abs/1`
+    * `binary_part/3`
+    * `bit_size/1`
+    * `byte_size/1`
+    * `div/2`
+    * `elem/2`
+    * `hd/1`
+    * `length/1`
+    * `map_size/1`
+    * `node/0`
+    * `node/1`
+    * `rem/2`
+    * `round/1`
+    * `self/0`
+    * `tl/1`
+    * `trunc/1`
+    * `tuple_size/1`
+  * the following handful of Erlang bitwise operations, if imported from the `Bitwise` module:
+    * `band/2` or the `&&&` operator
+    * `bor/2` or the `|||` operator
+    * `bnot/1` or the `~~~` operator
+    * `bsl/1` or the `<<<` operator
+    * `bsr/1` or the `>>>` operator
+    * `bxor/2` or the `^^^` operator
+
+## Defining custom guard expressions
```

Comment:
I want to add defguard to Elixir on v1.4 so we can keep this session around but it will likely change before 1.4 is out.